test/hotspot/jtreg/compiler/lib/ir_framework/IRNode.java

```diff
@@ -907,6 +907,22 @@ public class IRNode {
         beforeMatchingNameRegex(MUL, "Mul(I|L|F|D)");
     }
 
+    public static final String MUL_ADD_S2I = PREFIX + "MUL_ADD_S2I" + POSTFIX;
+    static {
+        beforeMatchingNameRegex(MUL_ADD_S2I, "MulAddS2I");
+    }
+
+    public static final String MUL_ADD_VS2VI = PREFIX + "MUL_ADD_VS2VI" + POSTFIX;
+    static {
+        superWordNodes(MUL_ADD_VS2VI, "MulAddVS2VI");
```

Comment:
Is there a reason this is not a `vectorNode`?

Reply:
First I've noticed that `MulAddVS2VI` is only created in Superword as far as I see. But now that [JDK-8310308](https://bugs.openjdk.org/browse/JDK-8310308) is in, I think I can switch to `vectorNode()`. Additionally, I think I need to revisit the rule for aarch64 to see if I can do better with the features from JDK-8310308. In the .ad file it says that it's only possible to use these nodes if the length is 16:
https://github.com/openjdk/jdk/blob/5350fd617390aaaedf8dd8821418c796cb1c38b3/src/hotspot/cpu/aarch64/aarch64_vector.ad#L182-L185